app/models/manageiq/providers/azure/cloud_manager/metrics_capture.rb

```diff
@@ -206,19 +257,36 @@ def resource_uri
     ].join('/')
   end
 
-  def api_counters_info
-    COUNTERS_INFO[:api]
+  def resource_url(tail_path = nil, query = {})
+    @resource_url ||= URI.join(@connection.environment.resource_url, resource_uri).freeze
+
+    url       = @resource_url.dup
+    url.path  = [url.path, tail_path].join('/') if tail_path
+    url.query = URI.encode_www_form(query) unless query.empty?
+
+    url
   end
 
-  def api_metric_names
-    METRIC_NAMES[:api]
+  def metrics_url
+    @metrics_url ||= resource_url(
+      # https://docs.microsoft.com/en-us/rest/api/monitor/metrics/list#uri-parameters
+      'providers/microsoft.insights/metrics',
+      'timespan'    => "#{@start_time}/#{@end_time}",
+      'interval'    => INTERVAL_1_MINUTE,
+      'metricnames' => METRIC_NAMES[:api].join(','),
+      'aggregation' => 'Average',
+      'api-version' => '2018-01-01'
+    ).freeze
   end
 
-  def raw_counters_info
-    COUNTERS_INFO[:raw]
+  def definitions_url
+    @definitions_url ||= resource_url(
+      'providers/microsoft.insights/metricDefinitions',
+      'api-version' => '2015-07-01'
+    ).freeze
   end
 
-  def raw_metric_names
-    METRIC_NAMES[:raw]
+  def parse_timestamp(timestamp)
+    Time.parse(timestamp).utc
```

Comment:
Hardly seems worth being a separate method. Is it too pervasive to drop?

Reply:
Yeah agreed seems like overkill

Reply:
Adam, if you remember recent PR to the API, where I fixed similar timestamp format issue - it was caused by a lack of a common unified approach. That's why I made this method here. It is used in all the places where it is needed, and makes sure we'll get the same timestamp format everywhere.

Reply:
So, it's more like an interface method then, ok. Maybe we could move this into the core class, but that's a separate discussion.

Reply:
maybe some stub into core, because each provider can have own timestamp format - which is something for subclassing. but yeah, it should be a part of a bigger picture